spec/support/helpers.rb

```diff
@@ -347,6 +347,15 @@ def simulate_platform(platform)
       ENV["BUNDLER_SPEC_PLATFORM"] = old if block_given?
     end
 
+    def simulate_ruby_version(version)
+      return if version == RUBY_VERSION
+      old = ENV["BUNDLER_SPEC_RUBY_VERSION"]
```

Comment:
Couldn't you move the `spec/support/hax.rb` logic around manipulating `Object`'s constants into the `simulate_ruby_version` method? So the fairly interesting/tricky logic's not distributed across two classes.